### PR TITLE
feat: Support for Dataproc on GKE

### PR DESCRIPTION
**Pull request by prauc**
Adds Support for Dataproc on GKE

Closes [#9094](https://github.com/hashicorp/terraform-provider-google/issues/9094)

If this PR is for Terraform, I acknowledge that I have:

- [X] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [X] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [X] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/third_party/terraform/tests) (for handwritten resources or update tests).
- [X] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [X] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
dataproc: Added Support for Dataproc on GKE in `google_dataproc_cluster` (GA only)
```

### CODE REVIEW

**Comment by prauc** [comment]
some missing features added after alignment call with customer

File: mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb
Comment (c2thorn) [inline mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb:222]: This seems to combine VirtualClusterConfig, GKEClusterConfig, GKENodePoolConfig and potentially more (it's hard to follow) from the API. We aim to be as in line as possible to the API schema and deviating from that could lead to challenges updating the resource in the future as schema changes are breaking changes. What is the reasoning here?

File: mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb
Comment (c2thorn) [inline mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb:286]: Child fields that are not `ForceNew` imply that they are updatable. If this is the case, a test step showing the updatability of these fields is preferred

**Review by c2thorn — CHANGES_REQUESTED** [review]
Have some high-level questions about the schema. Also, [the documentation](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/website/docs/r/dataproc_cluster.html.markdown) file should be updated with the new fields, but I'd hold off on that until confirmation on the schema level questions so you don't duplicate work

File: mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb
Comment (prauc) [reply mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb:221]: I was also very surprised, but the beta-package seems not to have this implement. VirtualClusterConfig does not exist in type Cluster for package v1beta2: https://pkg.go.dev/google.golang.org/api/dataproc/v1beta2#Cluster

File: mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb
Comment (prauc) [reply mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb:222]: makes sense! I wanted to simplify the structure and make it easier for users to use. But I get your point. I will change this according to the API

File: mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb
Comment (prauc) [reply mmv1/third_party/terraform/resources/resource_dataproc_cluster.go.erb:286]: Thanks for the hint! Will add tests to mentioned child fields

**Comment by c2thorn** [comment]
Ran the test manually since VCR tests are beta-only, and it passed. Just taking another look before merging

**Pull request merged**
